src/meta/src/hummock/compaction/overlap_strategy.rs

```diff
@@ -129,12 +132,7 @@ impl OverlapInfo for HashOverlapInfo {
     }
 
     fn update(&mut self, table: &SstableInfo) {
-        let other = KeyRange::from(table.key_range.as_ref().unwrap());
-        if let Some(range) = self.target_range.as_mut() {
-            range.full_key_extend(&other);
-            return;
-        }
-        self.target_range = Some(other);
+        self.infos.push(table.clone());
```

Comment:
Maybe a better idea is to record and merge vnodemap for every table.

Reply:
> Maybe a better idea is to record and merge vnodemap for every table.

that's incorrect. because key range of each table is different :-(

Reply:
> No. It's a map.
> 
> ```rust
> pub struct HashOverlapInfo {
>     infos: HashMap<u32, KeyRange>,
> }
> ```

info will have several SSTs. different SSTs have different key ranges. different SSTs have different tables. different tables have different bitmaps. so they can't be merged